### PR TITLE
Reformat quantum_chess package.

### PR DESCRIPTION
**Pull request by PawelPamula**
This change includes:
+ optimizing imports
+ fixing non-idiomatic code (comparing to None with 'is' operator, chained comparison)
+ type fixes (removing implicit type conversion)
+ formatting fixes
+ documentation fixes

### CODE REVIEW

**Review by dstrain115 — APPROVED** [review]
These look good.  What formatter were you using?  I used yapf.  I wanted to make sure my (or whomever) next commit doesn't undo all the formatting changes.

**Comment by PawelPamula** [comment]
> These look good. What formatter were you using? I used yapf. I wanted to make sure my (or whomever) next commit doesn't undo all the formatting changes.

I made another pass with yapf. All the previous formatting changes were compatible, all the new ones are introduced by yapf.

File: recirq/quantum_chess/quantum_board.py [hunk 1]
```diff
@@ -374,7 +374,7 @@ def unhook(self, qubit: cirq.Qid) -> cirq.Qid:
 
         # Replace operations using the qubit with the ancilla instead
         self.circuit = self.circuit.transform_qubits(lambda q: new_qubit
-                                                     if q == qubit else q)
+        if q == qubit else q)
```
Comment (dstrain115) [inline recirq/quantum_chess/quantum_board.py:377]: Is this line correct?  It looks weird.

**Review by dstrain115 — APPROVED** [review]

File: recirq/quantum_chess/quantum_board.py (hunk 1 above)
Comment (PawelPamula) [reply recirq/quantum_chess/quantum_board.py]: Good catch. yapf sometimes goes against PEP just to make it look nice, apparently it failed here. Switched it to a single line.

File: recirq/quantum_chess/quantum_moves.py [hunk 2]
```diff
@@ -57,13 +57,13 @@ def merge_move(s1: cirq.Qid, s2: cirq.Qid, t: cirq.Qid):
     two target qubits.
```
Comment (cantwellc) [inline recirq/quantum_chess/quantum_moves.py:57]:
```suggestion
    two source qubits.
```

**Review by cantwellc — CHANGES_REQUESTED** [review]
Just a suggested change on the merge move documentation. Otherwise lgtm

**Pull request merged**